src/wing/CCPACSWingShell.cpp

```diff
@@ -99,4 +108,86 @@ TiglLoftSide CCPACSWingShell::GetLoftSide() const
     throw CTiglError("Cannot determine loft side, this shell is neither lower nor upper shell of parent");
 }
 
+bool CCPACSWingShell::SparSegmentsTest(gp_Ax1 nNormal, gp_Pnt nTestPoint, TopoDS_Shape nSparSegments) const
+{
+    TopTools_IndexedMapOfShape faceMap;
+    TopExp::MapShapes(nSparSegments, TopAbs_FACE, faceMap);
+    double u_min = 0., u_max = 0., v_min = 0., v_max = 0.;
+
+    // for symmetry
+    nTestPoint.SetY(fabs(nTestPoint.Y()));
+
+    CTiglWingStructureReference wingStructureReference = m_parent->GetWingStructureReference();
+    for (int f = 1; f <= faceMap.Extent(); f++) {
+        TopoDS_Face loftFace = TopoDS::Face(faceMap(f));
+        BRepAdaptor_Surface surf(loftFace);
+        Handle(Geom_Surface) geomSurf = BRep_Tool::Surface(loftFace);
+
+        BRepTools::UVBounds(TopoDS::Face(faceMap(f)), u_min, u_max, v_min, v_max);
+        gp_Pnt midPnt = surf.Value(u_min + ((u_max - u_min) / 2), v_min + ((v_max - v_min) / 2));
+
+        gp_Pnt startPnt = surf.Value(u_min, v_min + ((v_max - v_min) / 2));
+        gp_Pnt endPnt   = surf.Value(u_max, v_min + ((v_max - v_min) / 2));
+
+        // if the U / V direction of the spar plane is changed
+
+        gp_Ax1 a1Test0   = gp_Ax1(startPnt, gp_Vec(startPnt, endPnt));
+        gp_Ax1 a1TestZ   = gp_Ax1(startPnt, gp_Vec(gp_Pnt(0., 0., 0.), gp_Pnt(0., 0., 1.)));
+        gp_Ax1 a1Test_mZ = gp_Ax1(startPnt, gp_Vec(gp_Pnt(0., 0., 0.), gp_Pnt(0., 0., -1.)));
+
+        if (a1Test0.Angle(a1TestZ) < (20.0 * (M_PI / 180.))) {
```

Comment:
@rmaierl Why the 20 degrees checks?

Reply:
Based on experience... So basically because it works

Reply:
Are these 20 degrees specific to spars, or would they work with general geometries?

I ask, because I try to figure out, if the function can be used more generally.